optaplanner-8-benchmarks/optaplanner-perf-benchmark/src/test/java/org/jboss/qa/brms/performance/SolverConfigFromXmlTest.java

```diff
@@ -0,0 +1,59 @@
+import org.jboss.qa.brms.performance.examples.travelingtournament.TravelingTournamentExample;
+import org.jboss.qa.brms.performance.examples.tsp.TravelingSalesmanExample;
+import org.jboss.qa.brms.performance.examples.vehiclerouting.VehicleRoutingExample;
+import org.jboss.qa.brms.performance.realworld.cloudbalancing.CloudBalanceBenchmark;
+import org.jboss.qa.brms.performance.realworld.conferencescheduling.ConferenceSchedulingBenchmark;
+import org.jboss.qa.brms.performance.realworld.nurserostering.NurseRosteringBenchmark;
+import org.jboss.qa.brms.performance.score.constraintstream.FlightCrewSchedulingConstraintStreamsBenchmark;
+import org.jboss.qa.brms.performance.score.constraintstream.TravelingTournamentConstraintStreamsBenchmark;
+import org.jboss.qa.brms.performance.score.constraintstream.VehicleRoutingConstraintStreamsBenchmark;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.optaplanner.core.config.constructionheuristic.ConstructionHeuristicType;
+import org.optaplanner.core.config.solver.SolverConfig;
+import org.optaplanner.core.config.solver.termination.TerminationConfig;
+
+class SolverConfigFromXmlTest {
+
+    private static final TerminationConfig oneSecondTermination = new TerminationConfig();
+
+    private static Stream<Arguments> problemSetAndSolverConfig() {
+        return Stream.of(
+                Arguments.of(new CloudBalanceBenchmark(CloudBalancingExample.DataSet.CB_100_300), Examples.CLOUD_BALANCING.getSolverConfigFromXml()),
+                Arguments.of(new ConferenceSchedulingBenchmark(ConferenceSchedulingExample.DataSet.TALKS_36_TIMESLOTS_12_ROOMS_5), Examples.CONFERENCE_SCHEDULING.getSolverConfigFromXml()),
+                Arguments.of(new FlightCrewSchedulingConstraintStreamsBenchmark(FlightCrewSchedulingExample.DataSet.EUROPE_175_FLIGHTS_7_DAYS), Examples.FLIGHT_CREW_SCHEDULING.getSolverConfigFromXml()),
+                Arguments.of(new NurseRosteringBenchmark(NurseRosteringExample.DataSet.SPRINT), Examples.NURSE_ROSTERING.getSolverConfigFromXml()),
+                Arguments.of(new TSPConstructionBenchmark(ConstructionHeuristicType.FIRST_FIT, TravelingSalesmanExample.DataSet.LU_980), Examples.TRAVELING_SALESMAN.getSolverConfigFromXml()),
+                Arguments.of(new TravelingTournamentConstraintStreamsBenchmark(TravelingTournamentExample.DataSet.SUPER_06), Examples.TRAVELING_TOURNAMENT.getSolverConfigFromXml()),
+                Arguments.of(new VehicleRoutingConstraintStreamsBenchmark(VehicleRoutingExample.DataSet.VRP_USA_100_10), Examples.VEHICLE_ROUTING.getSolverConfigFromXml())
+        );
+    }
+
+    @BeforeAll
+    public static void terminationAndInitSolution() {
+        oneSecondTermination.setSecondsSpentLimit(1L);
```

Comment:
I would rather use step-based termination:
- it's reproducible (on slower machines, the solver might not get to a point of failure in 1 second)
- doing just a couple of steps can actually take less than a second (something to try)

Reply:
Sorry, but I am not completely sure about that. StepCountTermination can only be used for phase termination. So some of the examples have createInitialSolution which returns initialized-calculated Solution and some of them returns just a problem [1]

So in case to use step limit I would need to add them as well that would requre  to change benchmarks. I do not need solver to get any of the score. I need to check here that the xml file for the solver was loaded to create the solver. So if my feelings are right that should be enough 

 [1] http://github.com/dupliaka/kie-benchmarks/blob/40b408917d908c462005068f6d6b99a9cb12df8e/optaplanner-8-benchmarks/optaplanner-perf-framework/src/main/java/org/jboss/qa/brms/performance/examples/travelingtournament/TravelingTournamentExample.java#L34-L34

Reply:
Initialization of the problem (either through a custom initializer or through running a CH phase) needs to happen regardless of what the termination condition is. But it's possible that running a CH will take more than a second.

Both cases are sufficient to check if the solver XML and the domain model are correct.